cmd/process-agent/collector.go

```diff
@@ -336,3 +339,32 @@ func isHTTPTimeout(err error) bool {
 	}
 	return false
 }
+
+func hasContainers(body []byte) bool {
+	m, err := model.DecodeMessage(body)
+	if err != nil {
+		log.Errorf("Unable to decode message body: %s", err)
+		return false
+	}
+	switch v := m.Body.(type) {
+	case *model.CollectorProc:
+		if v.GetContainers() != nil {
```

Comment:
I would write these are `len(v.GetContainers()) > 0`. It's possible for a slice to be non-nil and empty.